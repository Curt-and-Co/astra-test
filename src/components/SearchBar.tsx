import React, {FC} from 'react';
import {View, TextInput, StyleSheet} from 'react-native';

interface SearchBarProps {
  onSearch: (query: string) => void;
}

const SearchBar: FC<SearchBarProps> = ({onSearch}) => {
  return (
    <View style={styles.container}>
      <TextInput
        placeholder="Search TV Series"
        onChangeText={onSearch}
        style={styles.input}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 10,
  },
  input: {
    borderWidth: 1,
    borderColor: 'gray',
    borderRadius: 5,
    padding: 10,
  },
});

export default SearchBar;
