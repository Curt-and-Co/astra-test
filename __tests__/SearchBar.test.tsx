import React from 'react';
import renderer from 'react-test-renderer';
import SearchBar from '../src/components/SearchBar';

describe('SearchBar', () => {
  it('renders correctly', () => {
    const mockOnSearch = jest.fn();
    const tree = renderer
      .create(<SearchBar onSearch={mockOnSearch} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onSearch with the correct value when text changes', () => {
    const mockOnSearch = jest.fn();
    const instance = renderer.create(
      <SearchBar onSearch={mockOnSearch} />,
    ).root;

    const textInput = instance.findByType('TextInput');
    textInput.props.onChangeText('Test Query');

    // Check if onSearch was called with the correct value
    expect(mockOnSearch).toHaveBeenCalledWith('Test Query');
  });
});
