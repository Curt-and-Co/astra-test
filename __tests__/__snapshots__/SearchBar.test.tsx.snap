// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "marginHorizontal": 16,
    }
  }
>
  <TextInput
    onChangeText={[MockFunction]}
    placeholder="Search TV Series"
    style={
      {
        "backgroundColor": "white",
        "borderColor": "#fafafa",
        "borderRadius": 16,
        "borderWidth": 1,
        "flex": 1,
        "padding": 12,
      }
    }
  />
</View>
`;
